File: 1.数组/5.长度最小的子数组/209.长度最小的子数组.cpp
/*
 * @lc app=leetcode.cn id=209 lang=cpp
 *
 * [209] 长度最小的子数组
 */

// @lc code=start
class Solution {
    public:
        int minSubArrayLen(int target, vector<int>& nums) {
            int res = INT32_MAX;
            int sums = 0;
            int i = 0;
            for (int j = 0; j < nums.size(); j ++ )
            {
                sums += nums[j];
                while (sums >= target)
                {
                    int len = j - i + 1;
                    res = len < res ? len : res;
                    sums -= nums[i ++ ];
                }
            }
            return res == INT32_MAX ? 0 : res;
        }
    };
// @lc code=end

File: 面经150/3.滑动窗口/209.长度最小的子数组.py
#
# @lc app=leetcode.cn id=209 lang=python3
#
# [209] 长度最小的子数组
#
from mytools import *
# @lc code=start
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        j = 0
        n = len(nums)
        s = 0
        res = inf
        for i in range(n):
            s += nums[i]
            while j <= i and s >= target:
                res = min(res, i - j + 1)
                s -= nums[j]
                j += 1
        return res if res != inf else 0
# @lc code=end

